internal/modules/worklist.go

```diff
@@ -165,26 +166,19 @@ func (list *workList) push(task *resolveModule) {
 
 // getLatestModuleForConstraints returns the latest module that satisfies the provided constraints
 func (list *workList) getLatestModuleForConstraints(ctx context.Context, item *workItem, token cfg.SecretData) (*resolver.Version, error) {
-	constraints := make([]string, 0)
-	history := []resolution{}
-
 	m := item.spec
 	module := item.inProgressResolution
 
 	module.mu.Lock()
-	history = append(history, module.history...)
+	history := append([]resolution{}, module.history...)
 	module.mu.Unlock()
 	defer func() {
 		module.mu.Lock()
 		module.history = history
 		module.mu.Unlock()
 	}()
 
-	for _, r := range history {
-		if r.constraint != "" {
-			constraints = append(constraints, r.constraint)
-		}
-	}
+	constraints := getUniqueConstraints(history)
```

Comment:
suggestion: clearer name
```suggestion
	constraints := sortedUniqueConstraints(history)
```